include/swift/Driver/DependencyGraph.h

```diff
@@ -168,13 +175,20 @@ class DependencyGraphImpl {
     (void)newlyInserted;
   }
 
+  void
+  countMarking(CompilationCounters &counters,
+               DependencyMaskTy kinds, bool isCascading);
+
   void markTransitive(SmallVectorImpl<const void *> &visited,
-                      const void *node, MarkTracerImpl *tracer = nullptr);
+                      const void *node,
+                      CompilationCounters &counters,
```

Comment:
Can you put this inside MarkTracer instead? Right now DependencyGraphImpl is still pretty generic and I like that we'd be able to break it out for some other purpose if we had to.